### ISSUE TITLE
Azure Marketplace images for Debian 7/8 and CentOS 7.1 hang on lecture notice during converge

### ISSUE THREAD

**Issue by stuartpreston** [pendrica/kitchen-azurerm#19]
Test Kitchen 1.4.2
Net-ssh 3.0.2

.kitchen.yml
```
---
driver:
  name: azurerm

driver_config:
  subscription_id: 'b6e7eee9-YOUR-GUID-HERE-03ab624df016'
  location: 'West Europe'
  machine_size: 'Standard_DS3'
  storage_account_type: 'Premium_LRS'
  boot_diagnostics_enabled: 'false'

provisioner:
  name: chef_zero

platforms:
  - name: centos-7.1
    driver_config:
      image_urn: OpenLogic:CentOS:7.1:latest
  - name: ubuntu-15.10
    driver_config:
      image_urn: Canonical:UbuntuServer:15.10:latest
  - name: debian-7
    driver_config:
      image_urn: credativ:Debian:7:latest
      
verifier:
  name: inspec
            
suites:
  - name: cfgmgmtcamp
    run_list:
      - recipe[azure-linux::default]
    attributes:
```

**Comment by stuartpreston** [comment]
This is due to passwordless sudo only being supported with Ubuntu (which uses cloud-init) vs the others that use waagent to perform the config. Working on a PR to support SSH key access which should unblock this...

**Comment by stuartpreston** [comment]
Some changes for this have been pushed to branch **support-sshkey** in case anyone is interested in taking a look. I'll try and find some time to push a prerelease gem up soon.

**Comment by stuartpreston** [comment]
This is available in 0.3.0-pre
